interactive_ai/workflows/project_ie/job/usecases/data_redaction_usecase.py

```diff
@@ -409,27 +410,41 @@ def remove_lock_in_mongodb_doc(doc: dict) -> dict:
         return doc
 
     @staticmethod
-    def purge_model_docs_if_necessary(doc: dict) -> dict:
-        """
-        Purge model document metadata by marking it as purged and resetting specific fields.
-
-        :param doc: A dictionary representing a model document.
-        :return: The modified document after purging.
-        """
-        purge_info = doc.get("purge_info")
-        if purge_info is not None and purge_info["is_purged"]:
+    def purge_model_docs_if_necessary(include_model_ids: set[ObjectId]) -> Callable[[dict], dict]:
```

Comment:
```suggestion
    def purge_all_model_docs(model_ids_to_keep: set[ObjectId]) -> Callable[[dict], dict]:
```
- the "if_necessary" is unclear and doesn't make a good function name
- renamed the "include" parameter because it can be ambiguous (include in the export vs include in the purge).